liquibase-standard/src/main/java/liquibase/database/DatabaseFactory.java

```diff
@@ -230,6 +232,18 @@ public DatabaseConnection openConnection(String url,
 
             if(selectedDriverClass.contains("oracle")) {
               driverProperties.put("remarksReporting", "true");
+                // Try to split a pair CONNECT_USER[PROXY_USER] into two and store them in their properties
+                String userValue = driverProperties.getProperty("user");
+                if (userValue != null) {
+                    Pattern PROXY_USER_PATTERN = Pattern.compile("^([^\\[]+)\\[([^\\]]+)\\]$");
```

Comment:
This pattern is compiled every time openConnection() is called. Could you make it a static constant?